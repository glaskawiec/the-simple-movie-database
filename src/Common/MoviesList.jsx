import React from 'react';
import PropTypes from 'prop-types';
import MoviesWrapper from '../Routes/Discover/MoviesWrapper';
import Movie from '../Routes/Discover/Movie/Movie';
import LoadingBars from '../Routes/Discover/Movie/LoadingBars/LoadingBars';
import LoadingScreen from '../Routes/Discover/Movie/LoadingBars/LoadingScreen';

const formatDescription = (text) => {
  // @TODO: Correct format
  if (text.length > 200) {
    return `${text.substring(0, 200)}...`;
  }
  return text;
};


const MoviesList = ({ movies, isLoading }) => (
  <MoviesWrapper isLoading={isLoading}>
    {isLoading ? <LoadingBars /> : movies.map(movie => (
      <Movie
        key={movie.title}
        posterSrc={`https://image.tmdb.org/t/p/w185_and_h278_bestv2${movie.poster_path}`}
        description={formatDescription(movie.overview)}
        metaInformation={movie.release_date}
        title={movie.title}
      />
    ))}
  </MoviesWrapper>
);


MoviesList.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object).isRequired,
  isLoading: PropTypes.bool.isRequired,
};

export default MoviesList;
