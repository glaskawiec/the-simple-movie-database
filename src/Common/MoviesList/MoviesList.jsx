import React from 'react';
import PropTypes from 'prop-types';
import MoviesListWrapper from './MoviesListWrapper';
import Pagination from './Pagination/Pagination';
import LoadingBars from './Movie/LoadingBars/LoadingBars';
import Movie from './Movie/Movie';
import ErrorMessage from './Movie/ErrorMessage/ErrorMessage';

const formatDescription = (text) => {
  // @TODO: Correct format
  if (text.length > 150) {
    return `${text.substring(0, 150)}...`;
  }
  return text;
};

const mobileSrc = 'w1000_and_h563_face';
const desktopSrc = 'w185_and_h278_bestv2';

const MoviesList = ({
  movies, isLoading, totalPages, currentPage, onPageChange, isError,
}) => {
  const isMobile = window.innerWidth <= 768;
  let content;
  if (isError) {
    content = <ErrorMessage />;
  } else if (isLoading) {
    content = <LoadingBars />;
  } else {
    content = movies.map(movie => (
      <Movie
        key={movie.id}
        posterSrc={`https://image.tmdb.org/t/p/${isMobile ? mobileSrc : desktopSrc}${movie.poster_path}`}
        description={formatDescription(movie.overview)}
        metaInformation={movie.release_date}
        title={movie.title}
      />
    ));
  }
  return (
    <>
      <Pagination
        total={totalPages}
        onPageChange={onPageChange}
        current={currentPage}
      />
      <MoviesListWrapper
        withPagination={totalPages > 1}
        isError={isError}
        isLoading={isLoading}
      >
        {content}
      </MoviesListWrapper>
      <Pagination
        total={totalPages}
        onPageChange={onPageChange}
        current={currentPage}
      />
    </>
  );
};

MoviesList.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object).isRequired,
  isLoading: PropTypes.bool.isRequired,
  isError: PropTypes.bool.isRequired,
  totalPages: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired,
  onPageChange: PropTypes.func.isRequired,
};

export default MoviesList;
