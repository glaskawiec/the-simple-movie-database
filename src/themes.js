const Themes = {
  light: {
    colors: {
      primary: '#F2355B',
      secondary: '#161616',
      background: '#FFFFFF',
      text: '#000000',
      mobileMenuOverlay: '#000000',
      card: '#FFFFFF',
      themeToggleThumb: '#161616',
      paginationHover: '#CDCDCD',
    },
  },
  dark: {
    colors: {
      primary: '#F2355B',
      secondary: '#090909',
      background: '#161616',
      text: 'hsla(0,0%,100%,0.88)',
      mobileMenuOverlay: '#FFFFFF',
      card: '#090909',
      themeToggleThumb: '#FFFFFF',
      paginationHover: '#292929',
    },
  },
};

export default Themes;
