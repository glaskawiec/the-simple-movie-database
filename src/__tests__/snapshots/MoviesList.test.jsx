import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import themes from '../../themes';
import MoviesList from '../../common/moviesList/MoviesList';

test('MoviesList snapshot', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={themes.light}>
        <MoviesList
          currentPage={1}
          isError={false}
          isLoading={false}
          onPageChange={() => {}}
          totalPages={1}
        />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
