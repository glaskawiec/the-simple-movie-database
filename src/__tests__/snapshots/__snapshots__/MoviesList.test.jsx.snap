// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList snapshot 1`] = `
<div
  className="sc-bdVaJa fnhhZk"
/>
`;
